### PR TITLE
Finish startup sequence for process that exit too early

### PR DESCRIPTION
**Pull request by mariospr**
Add a new public method in ShellWindowTracker to wrap sn_startup_sequence_complete() and use that to make sure that these kind of applications get their startup sequences completed, and then properly moved to STOPPED.

https://phabricator.endlessm.com/T22416

### CODE REVIEW

File: src/shell-window-tracker.c [hunk 1]
```diff
@@ -1008,6 +1008,11 @@ shell_startup_sequence_create_icon (ShellStartupSequence *sequence, guint size)
   return texture;
```
Comment (smspillaz) [inline src/shell-window-tracker.c]: Typo in the commit message s/publich/public/

File: src/shell-app.c [hunk 2]
```diff
@@ -1253,12 +1253,37 @@ app_child_setup (gpointer user_data)
...
+static void
+_shell_app_watch_callback (GPid      pid,
+                           gint      status,
+                           ShellApp *app)
+{
+  if (app->state == SHELL_APP_STATE_STARTING)
+    {
+      ShellWindowTracker *tracker = shell_window_tracker_get_default ();
+      GSList *startup_sequences = shell_window_tracker_get_startup_sequences (tracker);
+      GSList *iter = NULL;
```
Comment (smspillaz) [inline src/shell-app.c]: `iter` can probably assigned with `startup_sequences` directly here

File: src/shell-app.c [hunk 3]
```diff
@@ -1305,7 +1330,8 @@ shell_app_launch (ShellApp     *app,
 #else
                                                    NULL, NULL,
 #endif
-                                                   wait_pid, NULL,
+                                                   (GDesktopAppLaunchCallback) wait_pid,
```
Comment (smspillaz) [inline src/shell-app.c]: Is this change relevant to this commit?

**Comment by smspillaz** [comment]
Looks good - just left a few drive-by nitpicks

**Comment by mariospr** [comment]
> Looks good - just left a few drive-by nitpicks

Thanks. I've addresed the typo in the commit message but I don't think there' nothing that needs to be done with regard the other two comments, see my answers inline. Please let me know what you think.

**Comment by mariospr** [comment]
As mentioned in the ticket, we will be fixing this in GTK after all, so closing this MR now.

Let's continue in https://gitlab.gnome.org/GNOME/gtk/issues/1084
